pt/santander.py

```diff
@@ -0,0 +1,411 @@
+            if formatted_phone != d["contact:phone"]:
+                d["contact:phone"] = formatted_phone
+
+        if "contactData" in nd:
+            phones = []
+            if "phoneNumber" in nd["contactData"]:
+                phone_formatted = format_phonenumber(nd["contactData"]["phoneNumber"])
+                if phone_formatted not in d["contact:phone"].split(";"):
+                    d["contact:phone"] += ";" + phone_formatted
+            if nd["contactData"]["customerPhone"] != "":
+                phone_formatted = format_phonenumber(nd["contactData"]["customerPhone"])
+                if phone_formatted not in d["contact:phone"].split(";"):
+                    d["contact:phone"] += ";" + phone_formatted
+            if nd["contactData"]["fax"] != "":
+                d["contact:fax"] = format_phonenumber(nd["contactData"]["fax"])
+            if nd["contactData"]["email"] not in d["contact:email"].split(";"):
+                d["contact:email"] = nd["contactData"]["email"]
+
+        if "socialData" in nd:
+            d["contact:youtube"] = nd["socialData"].get("youtubeLink", "")
+            d["contact:facebook"] = nd["socialData"].get("facebookLink", "")
+            d["contact:twitter"] = nd["socialData"].get("twitterLink", "")
+            d["contact:linkedin"] = nd["socialData"].get("linkedinLink", "")
+            d["contact:instagram"] = nd["socialData"].get("instagramLink", "")
+            d["contact:tiktok"] = nd["socialData"].get("tiktokLink", "")
+            
+        # Schedule
+        opening_hours = ""
+        schedule = nd["schedule"]
+        workingDays = {}
+        
+        for workday in schedule["workingDay"]:
+            wd = WEEKDAY2STR[workday]
+            if schedule["workingDay"][workday] == []:
+                continue
+            workingHours = ",".join(fixWorkingHours(schedule["workingDay"][workday]))
+            if workingHours == "Encerrado":
+                continue
+            if workingHours not in workingDays:
+                workingDays[workingHours] = []
+            workingDays[workingHours].append(wd)
+        for workingHours, days in workingDays.items():
+            days.sort(key=lambda s: DAYS.index(s))
+            if opening_hours != "":
+                opening_hours += "; "
+            days = merge_weekdays(days)
+            opening_hours += ",".join(days) + " " + workingHours
+        days_off = []
+        for day in schedule["specialDay"]:
+            if day["time"] == [None]:
+                month, day = day["date"].split("-")
+                monthName = MONTH_NAMES[month]
+                days_off.append(f"{monthName} {day}")
+            else:
+                opening_hours = f"<ERR:{json.dump(schedule)}>"
+                days_off = []
+                break
+        if days_off != []:
+            opening_hours += "; " + ",".join(days_off) + " off"
+        d["opening_hours"] = opening_hours
```

Comment:
Two more tags to consider: every script sets `source:contact` and `source:opening_hours` to `website` (some of them do this conditionally, but they should just always set them and I'm going to change that). I'm adding them so that a) the origin is clear, and b) I know that nobody else has touched the values (unless they didn't change `website` to `survey`, which happens sometimes) and so I'm free to update them next time when they change on the website.